core/bentley/src/BentleyError.ts

```diff
@@ -658,3 +658,60 @@ export class BentleyError extends Error {
     }
   }
 }
+
+function hasProperty<T extends string>(obj: object, propertyName: T): obj is { [key in T]: unknown } {
+  return propertyName in obj;
+}
+
+/** @internal */
+export function getErrorStack(error: unknown): string | undefined {
+  if (typeof error === "object" && error !== null && hasProperty(error, "stack") && typeof error.stack === "string")
+    return error.stack;
+
+  return undefined;
+}
+
+/** @internal */
+export function getErrorMessage(error: unknown): string | undefined {
+  if (typeof error === "string")
+    return error;
+
+  if (error instanceof Error)
+    return error.toString();
+
+  if (typeof error === "object") {
+    if (error === null)
+      return undefined;
+
+    if (hasProperty(error, "message") && typeof error.message === "string")
```

Comment:
i don't think `hasProperty` is necessary.

Reply:
@kabentley my latest changes here are using a different custom type guard that should be a bit less obnoxious while still using `error: unknown` for the `error` parameter.  Does that look OK to you?